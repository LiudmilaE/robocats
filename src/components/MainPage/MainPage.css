.MainPage {
  text-align: center;
}

.MainPage-header {
  background-color: #4F576E;
  /* height: 210px; */
  padding: 20px;
  color: white;
}

.MainPage-title {
  font-size: 2.5rem;
  text-transform: uppercase;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}

.MainPage-intro {
  font-size: large;
}
