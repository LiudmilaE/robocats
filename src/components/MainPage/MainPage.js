import React, { Component } from 'react';
import { CardsList } from '../UI/CardsList';
import { SearchField } from '../SearchField/SearchField';
import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';
import './MainPage.css';

class MainPage extends Component {

  componentDidMount() {
    this.props.onRequestRobocats()
  }

  filterRobocats = () => {
      const { robocats, searchField } = this.props;
    return robocats.filter(cat => cat.username.includes(searchField));
  }

  render() {
    const { onSearchChange, isPending } = this.props;
    
    if(isPending) {
      return <h2>Loading</h2>
    }
    return (
      <div className="MainPage MainPage-container">
        <header className="MainPage-header">
          <h1 className="MainPage-title">Robocats</h1>
          <h2>Here you can find a new cute pet!</h2>
          <SearchField onSearch={onSearchChange}/>
        </header>
        <ErrorBoundary>
          <CardsList data={this.filterRobocats()}/>
        </ErrorBoundary>
      </div>
    );
  }
}

export default MainPage;