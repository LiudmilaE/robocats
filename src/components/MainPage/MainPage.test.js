import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

describe("MainPage", () => {
    let wrapper;
    beforeEach(() => {
        const mockProps = {
            onRequestRobocats: jest.fn(),
            searchField: '',
            onSearchChange: jest.fn(),
            robocats: [],
            isPending: false
        };
        wrapper = shallow(<MainPage {...mockProps}/>)
    });

    it('renders MainPage without crashing', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders Loading status if isPending: true', () => {
        const mockProps = {
            onRequestRobocats: jest.fn(),
            isPending: true
        };
        wrapper = shallow(<MainPage {...mockProps}/>);
        expect(wrapper.text()).toEqual('Loading');
    });

    it('filter robocats correctly', () => {
        expect(wrapper.instance().filterRobocats()).toEqual([]);

        const mockProps2 = {
            onRequestRobocats: jest.fn(),
            onSearchChange: jest.fn(),
            robocats: [{username: 'foo', name: "Foo Baz Bar"}],
            searchField: "fo",
            isPending: false
        }
        const wrapper2 = shallow(<MainPage {...mockProps2}/>);
        expect(wrapper2.instance().filterRobocats())
            .toEqual([{username: 'foo', name: "Foo Baz Bar"}]);

        const mockProps3 = {
            onRequestRobocats: jest.fn(),
            onSearchChange: jest.fn(),
            robocats: [{username: 'foo', name: "Foo Baz Bar"}],
            searchField: "ho",
            isPending: false
        }
        const wrapper3 = shallow(<MainPage {...mockProps3}/>);
        expect(wrapper3.instance().filterRobocats())
            .toEqual([]);
    });
});
