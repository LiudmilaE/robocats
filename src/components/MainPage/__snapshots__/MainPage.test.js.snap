// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders MainPage without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    isPending={false}
    onRequestRobocats={[Function]}
    onSearchChange={[Function]}
    robocats={Array []}
    searchField=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="MainPage-header"
>
          <h1
                    className="MainPage-title"
          >
                    Robocats
          </h1>
          <h2>
                    Here you can find a new cute pet!
          </h2>
          <SearchField
                    onSearch={[Function]}
          />
</header>,
        <ErrorBoundary>
          <Unknown
                    data={Array []}
          />
</ErrorBoundary>,
      ],
      "className": "MainPage MainPage-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="MainPage-title"
>
              Robocats
</h1>,
            <h2>
              Here you can find a new cute pet!
</h2>,
            <SearchField
              onSearch={[Function]}
/>,
          ],
          "className": "MainPage-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Robocats",
              "className": "MainPage-title",
            },
            "ref": null,
            "rendered": "Robocats",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Here you can find a new cute pet!",
            },
            "ref": null,
            "rendered": "Here you can find a new cute pet!",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onSearch": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Unknown
            data={Array []}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="MainPage-header"
>
            <h1
                        className="MainPage-title"
            >
                        Robocats
            </h1>
            <h2>
                        Here you can find a new cute pet!
            </h2>
            <SearchField
                        onSearch={[Function]}
            />
</header>,
          <ErrorBoundary>
            <Unknown
                        data={Array []}
            />
</ErrorBoundary>,
        ],
        "className": "MainPage MainPage-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="MainPage-title"
>
                Robocats
</h1>,
              <h2>
                Here you can find a new cute pet!
</h2>,
              <SearchField
                onSearch={[Function]}
/>,
            ],
            "className": "MainPage-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Robocats",
                "className": "MainPage-title",
              },
              "ref": null,
              "rendered": "Robocats",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Here you can find a new cute pet!",
              },
              "ref": null,
              "rendered": "Here you can find a new cute pet!",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onSearch": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Unknown
              data={Array []}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "data": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
